File: src/main/java/fr/lewon/web/bot/entities/input/shop/ActionBuyItem.java
package fr.lewon.web.bot.entities.input.shop;

import fr.lewon.bot.http.body.urlencoded.FUEMember;
import fr.lewon.web.bot.entities.input.Action;
import fr.lewon.web.bot.entities.shop.Item;

public class ActionBuyItem extends Action {

	@FUEMember("id_item")
	private String idItem;
	
	@FUEMember
	private String type;
	
	@FUEMember
	private Integer who = 1;
	
	public ActionBuyItem(Item item) {
		super("Item", "buy");
		this.idItem = item.getId();
		this.type = item.getType();
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/shop/ActionBuyItem.kt
package fr.lewon.bot.hh.entities.input.shop

import com.fasterxml.jackson.annotation.JsonProperty
import fr.lewon.bot.hh.entities.input.Action
import fr.lewon.bot.hh.entities.shop.Item

class ActionBuyItem(item: Item) : Action("Item", "buy") {
    @JsonProperty("id_item")
    val idItem: String? = item.id
    @JsonProperty
    val type: String? = item.type
    @JsonProperty
    val who = 1

}